### PR TITLE
[docs] Update docs and add rust guideline

### PR DESCRIPTION
**Pull request by mssun**
## Description

- Update our threat model
- Add Rust development guideline

### CODE REVIEW

File: docs/threat-model.md [hunk 1]
```diff
@@ -25,17 +29,24 @@ from company B, and use the deep learning inference engine provided by company
...
+attestation service (e.g., Intel Attestation Service). Private data are securely
+provisioned to the Teaclave enclave only if the attestation passes. After the
+provision, no privileged software is able to access the memory content owned by
+the enclave from outside.
 
 The remote attestation functionality implemented by Teaclave is augmented from
-the method described by an Intel [white paper](https://arxiv.org/abs/1801.05863).
-The complicated structure of Teaclave requires additional work for remote
-attestation, which is explained in details via a separate
+the method described by an [white paper](https://arxiv.org/abs/1801.05863). In a
```
Comment (m4sterchain) [inline docs/threat-model.md:43]: an -> a

File: docs/threat-model.md [hunk 2]
```diff
@@ -25,17 +29,24 @@ from company B, and use the deep learning inference engine provided by company
...
+nutshell, enclaves in Teaclave will establish trusted channel on attested TLS
+for communication. The complicated structure of Teaclave requires additional
+work for remote attestation, which is explained in details via a separate
 [documentation](mutual_attestation.md).
+
+Side channels are out of scope for current Teaclave's implementation. While we
+acknowledge that existing enclaves may be vulnerable to various kinds of side
+channel attacks, we will try our best to mitigate existing attacks by adopt
```
Comment (m4sterchain) [inline docs/threat-model.md:51]: adopting

File: docs/rust-guideline.md [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+hold.
+
+## Error Handling
+
+Using `unwrap` or `expect` to get a value from an optional type may introduce
+runtime panic. Therefore, properly using the error handling mechanism provided
+by Rust can make the system robust and clean. In some cases, optional value can
+never be `None` internally, `unwrap` can be used with a comment explaining the
+assumptions and reasons. The same rule also applies to `panic` and similar
+functions which may cause runtime panic.
```
Comment (m4sterchain) [inline docs/rust-guideline.md:43]: Can we add some guidelines regarding using `unwrap` for test cases?

**Comment by mssun** [comment]
Fixed.

**Pull request merged**